VideoLocker/src/main/java/org/edx/mobile/util/Config.java

```diff
@@ -103,30 +108,30 @@ public boolean isEnabled() {
 
     /**
      * Course Enrollment configuration.
+     *
+     * If TYPE is not "webview", defaults to "native"
      */
     public class EnrollmentConfig {
-        private @SerializedName("ENABLED") boolean mEnabled;
-        private @SerializedName("NATIVE_COURSE_DISCOVERY_ENABLED") boolean mNativeCourseDiscoveryEnabled;
+        private @SerializedName("TYPE") String mCourseEnrollmentType;
         private @SerializedName("COURSE_SEARCH_URL") String mSearchUrl;
-        private @SerializedName("EXTERNAL_COURSE_SEARCH_URL") String mExternalSearchUrl;
         private @SerializedName("COURSE_INFO_URL_TEMPLATE") String mCourseInfoUrlTemplate;
 
-        public boolean isEnabled() {
-            return mEnabled;
-        }
-
-        public boolean isNativeCourseDiscoveryEnabled() {
-            return mNativeCourseDiscoveryEnabled;
+        public boolean isWebviewCourseDiscoveryEnabled() {
+            if (mCourseEnrollmentType != null) {
+                if (mCourseEnrollmentType.equalsIgnoreCase(EnrollmentType.WEBVIEW.name())) {
+                    return true;
+                } else if (!mCourseEnrollmentType.equalsIgnoreCase(EnrollmentType.NATIVE.name())) {
+                    // logger.debug("No match in config for COURSE_ENROLLMENT.TYPE:" + mCourseEnrollmentType + ". Defaulting to Native");
```

Comment:
So the problem is EnrollmentConfig is an inner class and expects to have a pointer to the outer Config class.

However, when `EnrollmentConfig` is created by Gson, it has no `this` pointer, so trying to access `Config.this.logger` is an NPE because `Config.this` is null, not because logger is null.